### PR TITLE
feat(msd): metrics

### PR DESCRIPTION
**Pull request by NikolaMilosa**
In this PR we introduce all [axum-otel-metrics](https://github.com/ttys3/axum-otel-metrics) alongside 4 custom metrics:
* `msd_definitions_load_errors_total`: number of errors while loading new targets per definition.
* `msd_definitions_load_successful`: status metric which indicates if last load was successful (1) or not (0).
* `msd_definitions_sync_errors_total`: number of errors while syncing the registry per definition.
* `msd_definitions_sync_successful`:  status metric which indicates if last scrape was successful (1) or not (0).

### CODE REVIEW

**Review by sasa-tomic — APPROVED** [review]
LGTM!

**Review by DFINITYManu — CHANGES_REQUESTED** [review]
Please fix or verify metrics names are correct (otel may apply some postprocessing to standardize).  Stand by for link.

https://prometheus.io/docs/practices/naming/ please follow this standard.

File: rs/ic-observability/multiservice-discovery/src/metrics.rs [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+impl MSDMetrics {
+    pub fn new() -> Self {
+        let meter = global::meter("axum-app");
+        let definitions = meter
+            .i64_up_down_counter("msd.definitions.total")
```
Comment (DFINITYManu) [inline rs/ic-observability/multiservice-discovery/src/metrics.rs:22]: Not a monotonically increasing counter, so total doesn't apply.  This one can be left at msd.definitions.

Comment (DFINITYManu) [reply rs/ic-observability/multiservice-discovery/src/metrics.rs:22]: Idem in other gauges.

https://opentelemetry.io/docs/specs/otel/metrics/api/#updowncounter

**Review by DFINITYManu — CHANGES_REQUESTED** [review]

File: rs/ic-observability/multiservice-discovery/src/metrics.rs [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+        Self::new()
+    }
+}
+
+impl MSDMetrics {
+    pub fn new() -> Self {
+        let meter = global::meter("axum-app");
+        let definitions = meter
+            .i64_up_down_counter("msd.definitions")
+            .with_description("Total number of definitions that multiservice discovery is scraping")
```
Comment (sasa-tomic) [inline rs/ic-observability/multiservice-discovery/src/metrics.rs:26]:
```suggestion
            .with_description("Total number of definitions that multiservice discovery is serving")
```
targets ==> scraping
definitions ==> serving
or am I wrong?

**Review by sasa-tomic — APPROVED** [review]

File: rs/ic-observability/multiservice-discovery/src/metrics.rs (hunk 2 above)
Comment (NikolaMilosa) [reply rs/ic-observability/multiservice-discovery/src/metrics.rs:26]: Definition == network, we scrape networks
1 network has many targets, we serve targets on /targets

**Comment by sasa-tomic** [comment]
> MSD Diff
> 
> * targets.json

Hmm this seems to be off.

**Comment by pietrodimarco-dfinity** [comment]
> ```diff
> 2001:ab8:1000:f81:6801:4fff:fe8c:5e50
> ```



> > MSD Diff
> > 
> > * targets.json
> 
> Hmm this seems to be off. cc @pietrodimarco-dfinity any idea why this diff is non-empty?

yes :) the registry got an update!
From main
Feb 09 09:26:55.783 DEBG Latest registry version: 41046

From branch
Feb 09 09:26:55.783 DEBG Latest registry version: 41049

**Review by DFINITYManu — APPROVED** [review]

**Pull request merged**